OgcApi.Net/DataProviders/SqlDataProvider.cs

```diff
@@ -493,5 +500,52 @@ public void DeleteFeature(string collectionId, string featureId, string apiKey =
         protected abstract IFeaturesSqlQueryBuilder GetFeaturesSqlQueryBuilder(SqlFeaturesSourceOptions collectionOptions);
 
         protected abstract Geometry ReadGeometry(DbDataReader dataReader, int ordinal, SqlFeaturesSourceOptions collectionSourceOptions);
+
+        public Task<byte[]> GetTileAsync(string collectionId, int tileMatrix, int tileRow, int tileCol, string datetime = null, string apiKey = null)
+        {
+            var bbox = CoordinateConverter.TileBounds(tileRow, tileCol, tileMatrix);
+            var features = GetFeatures(collectionId, bbox: bbox);
+
+            var vectorTile = new VectorTile();
+            var layer = new Layer();
+
+            foreach (var feature in features)
+            {
+                layer.Features.Add(feature);
+            }
+
+            vectorTile.Layers.Add(layer);
+
+            using var memoryStream = new MemoryStream();
+            vectorTile.Write(memoryStream);
+            memoryStream.Seek(0, SeekOrigin.Begin);
+
+            return Task.FromResult(memoryStream.ToArray());
+        }
+
+        public List<TileMatrixLimits> GetLimits(string collectionId)
+        {
+            var collectionOptions = (CollectionOptions)CollectionsOptions.GetSourceById(collectionId);
+            if (collectionOptions == null)
+            {
+                Logger.LogTrace(
+                    $"The source collection with ID = {collectionId} was not found in the provided options");
+                throw new ArgumentException($"The source collection with ID = {collectionId} does not exists");
+            }
+
+            var result = new List<TileMatrixLimits>();
+            for (var i = 0; i <= 22; ++i)
```

Comment:
In the for loop, replace prefix increment with postfix increment. In this context it makes no sense to use prefix increment.